packages/modules/promotion/src/models/campaign.ts

```diff
@@ -1,109 +1,36 @@
-import { DAL } from "@medusajs/framework/types"
-import {
-  DALUtils,
-  Searchable,
-  createPsqlIndexStatementHelper,
-  generateEntityId,
-} from "@medusajs/framework/utils"
-import {
-  BeforeCreate,
-  Collection,
-  Entity,
-  Filter,
-  OnInit,
-  OneToMany,
-  OneToOne,
-  OptionalProps,
-  PrimaryKey,
-  Property,
-  Rel,
-} from "@mikro-orm/core"
+import { model } from "@medusajs/framework/utils"
 import CampaignBudget from "./campaign-budget"
 import Promotion from "./promotion"
 
-type OptionalRelations = "budget"
-type OptionalFields =
-  | "description"
-  | "starts_at"
-  | "ends_at"
-  | DAL.SoftDeletableModelDateColumns
-
-const tableName = "promotion_campaign"
-const CampaignUniqueCampaignIdentifier = createPsqlIndexStatementHelper({
-  tableName,
-  columns: ["campaign_identifier"],
-  unique: true,
-  where: "deleted_at IS NULL",
-})
-
-@Entity({ tableName })
-@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
-export default class Campaign {
-  [OptionalProps]?: OptionalFields | OptionalRelations
-
-  @PrimaryKey({ columnType: "text" })
-  id!: string
-
-  @Searchable()
-  @Property({ columnType: "text" })
-  name: string
-
-  @Searchable()
-  @Property({ columnType: "text", nullable: true })
-  description: string | null = null
-
-  @Property({ columnType: "text" })
-  @CampaignUniqueCampaignIdentifier.MikroORMIndex()
-  campaign_identifier: string
-
-  @Property({
-    columnType: "timestamptz",
-    nullable: true,
+const Campaign_ = model
+  .define<any, any>(
```

Comment:
q: is the `<any, any>` type here intended?

Reply:
Yup. Fixed 👍